README.md

```diff
@@ -1 +1,56 @@
 ## Dockerised reverse proxy
+
+[![Build Status](https://travis-ci.com/reside-ic/proxy-nginx.svg?branch=master)](https://travis-ci.com/reside-ic/proxy-nginx)
+
+This repository contains a support for running a `nginx` proxy in a docker container in order to secure a web application.  It makes a number of assumptions:
+
+* You want to secure the app with TLS and redirect all http traffic to https
+* You have a single service container to proxy and it is speaking http
+
+The configuration takes as a starting point [`montagu-proxy`](https://github.com/vimc/montagu-proxy), which was adapted for use with [`orderly-web-deploy`](https://github.com/vimc/orderly-web-deploy).
+
+### Configuration
+
+Before starting we need to know what we are proxying (i.e., the name of the container on the docker network that is the main entrypoint) and what the proxy will be seen as to the outside world (the hostname, and ports for http and https).  The entrypoint takes these four values as arguments.
+
+### SSL Certificates
+
+The server will not start until the files `/run/proxy/certificate.pem` and `/run/proxy/key.pem` exist - you can get these into the container however you like; the proxy will poll for them and start within a second of them appearing.
+
+### Self signed certificate
+
+For testing it is useful to use a self-signed certificate.  These are not in any way secure.  To generate a self-signed certificate, there is a utility in the proxy container `self-signed-certificate` that will generate one on demand after receiving key components of the CSR.
+
+There is a self-signed certificate in the repo for testing generated with (on metal)
```

Comment:
Took me a while to parse this sentence. Maybe "There is a self-signed certificate in the repo for testing. This was generated with (on metal)"